semgrep/semgrep/core_output.py

```diff
@@ -298,30 +300,6 @@ def rule_matches(self, rules: List[Rule]) -> Dict[Rule, List[RuleMatch]]:
 
         For now assumes that all matches encapsulated by this object are from the same rulee
         """
-        # This will remove matches that have the same range but different
```

Comment:
You should probably keep some of this docstring so it's clear why they need to be in reverse

Reply:
I'm confused by this comment and actually didn't reference it while implementing the code. The comment makes it sound like it tries to deduplicate findings so that the largest bound metavariable is kept, but the `uniq_id` function actually includes the interpolated messages. So with two findings where one is bound to a "smaller" metavariable and the other is bound to a "larger" metavariable, there is no decision to be made about which one to discard: we keep them both because their messages are different.